viewer/src/logic/commands.ts

```diff
@@ -0,0 +1,35 @@
+import Engine, { Faction, GaiaHex, HighlightHex, Operator, Resource, Reward } from "@gaia-project/engine";
+import AvailableCommand, { AvailableHex } from "@gaia-project/engine/src/available-command";
+import { ButtonWarning, HighlightHexData } from "../data";
+
+export function hexMap(engine: Engine, coordinates: AvailableHex[]): HighlightHexData {
+  return new Map<GaiaHex, HighlightHex>(
+    coordinates.map((coord) => [
+      engine.map.getS(coord.coordinates),
+      {
+        cost: coord.cost,
+        warnings: coord.warnings,
+      },
+    ])
+  );
+}
+
+export function buttonWarning(message?: string): ButtonWarning | null {
+  return message == null ? null : { title: "Are you sure?", body: [message] };
```

Comment:
```suggestion
  return message && { title: "Are you sure?", body: [message] };
```

Just posting that once as an example of a JS trick: A && B will return A if !A, or B if !!A.